src/bucket/BucketManagerImpl.cpp

```diff
@@ -1213,6 +1221,137 @@ BucketManagerImpl::mergeBuckets(HistoryArchiveState const& has)
     return out.getBucket(*this, /*shouldSynchronouslyIndex=*/false);
 }
 
+static void
+processArchivalMetrics(
+    std::shared_ptr<Bucket const> const b,
+    UnorderedMap<LedgerKey, StateArchivalMetric>& ledgerEntries,
+    UnorderedMap<LedgerKey, StateArchivalMetric>& ttls)
+{
+    for (BucketInputIterator in(b); in; ++in)
+    {
+        auto const& be = *in;
+        bool isDead = be.type() == DEADENTRY;
+        LedgerKey k = isDead ? be.deadEntry() : LedgerEntryKey(be.liveEntry());
+        bool isTTL = k.type() == TTL;
+
+        if (!isTemporaryEntry(k) && !isTTL)
+        {
+            continue;
+        }
+
+        auto proccessEntry = [&](auto& map) {
+            auto iter = map.find(k);
+            if (iter == map.end())
+            {
+                StateArchivalMetric metric;
+                metric.le =
```

Comment:
I think this can get pretty expensive memory-wise if we're dealing with a production bucketlist with a large number of temp entries (in order of GBs). Perhaps we should only store the size and expiration ledger of LEs?

Reply:
I think we should also keep track of whether the newest version of a key is DEADENTRY or LIVEENTRY, but you're right, storing the whole LedgerEntry is unnecessary,